req2flatpak.py

```diff
@@ -705,10 +728,31 @@ def main():
     parser = cli_parser()
     options = parser.parse_args()
 
+    def _get_yaml_module_or_exit():
+        try:
+            # optional dependency, not imported at top
+            import yaml
+        except ImportError:
+            parser.error(
+                "Outputing YAML requires 'pyyaml' package: try 'pip install pyyaml'"
+        )
+        return yaml
+
```

Comment:
I don't oppose adding this function although I did wonder why you don't add another if-statement:

```py
if options.yaml:
    try:
        # optional dependency, not imported at top
        import yaml
    except ImportError:
        parser.error(
            "Outputing YAML requires 'pyyaml' package: try 'pip install pyyaml'"
    )

Reply:
seemed like we need this functionality in two places (and maybe three, although I'm not sure how to wire it into the classmethod).

But sure I can inline it.

Reply:
nm, I see another way

Reply:
> seemed like we need this functionality in two places

When using `parser` the function doesn't bring much value since that can be achieved with the single if-statement too.

> I'm not sure how to wire it into the classmethod

I suggest that the `FlatpakGenerator` API just forwards the ImportError, maybe with a more descriptive error message.